examples/data_type2.rs

```diff
@@ -0,0 +1,107 @@
+use raw::KeyType;
+use std::collections::HashMap;
+use std::os::raw::c_void;
+use std::ptr::null;
+use valkey_module::alloc::ValkeyAlloc;
+use valkey_module::digest::Digest;
+use valkey_module::key::{ValkeyKey, ValkeyKeyWritable};
+use valkey_module::native_types::ValkeyType;
+use valkey_module::{raw, valkey_module, Context, NextArg, ValkeyResult, ValkeyString};
+
+#[derive(Debug)]
+struct MyType {
+    data: i64,
+}
+
+static MY_VALKEY_TYPE: ValkeyType = ValkeyType::new(
+    "mytype123",
+    0,
+    raw::RedisModuleTypeMethods {
+        version: raw::REDISMODULE_TYPE_METHOD_VERSION as u64,
+        rdb_load: None,
+        rdb_save: None,
+        aof_rewrite: None,
+        free: Some(free),
+        digest: Some(digest),
+
+        // Currently unused by Redis
+        mem_usage: None,
+
+        // Aux data
+        aux_load: None,
+        aux_save: None,
+        aux_save2: None,
+        aux_save_triggers: 0,
+
+        free_effort: None,
+        unlink: None,
+        copy: None,
+        defrag: None,
+
+        copy2: None,
+        free_effort2: None,
+        mem_usage2: None,
+        unlink2: None,
+    },
+);
+
+unsafe extern "C" fn free(value: *mut c_void) {
+    drop(Box::from_raw(value.cast::<MyType>()));
+}
+
+unsafe extern "C" fn digest(md: *mut raw::RedisModuleDigest, value: *mut c_void) {
+    let mut dig = Digest::new(md);
+    let val = &*(value.cast::<MyType>());
+    dig.add_long_long(val.data);
```

Comment:
can these 3 lines be added to `digest` in `data_type.rs`?  That would DRY things up